web/packages/core/src/load-ruffle.ts

```diff
@@ -27,13 +32,51 @@ async function fetchRuffle(config: Config): Promise<typeof Ruffle> {
     // libraries, if needed.
     setPolyfillsOnLoad();
 
+    const extensionsSupported: boolean = (
+        await Promise.all([
+            bulkMemory(),
+            simd(),
+            saturatedFloatToInt(),
+            signExtensions(),
+        ])
+    ).every(Boolean);
+
+    if (extensionsSupported) {
+        console.log(
+            "All necessary WASM extensions are available, loading the fancy WASM module"
+        );
+    } else {
+        console.log(
+            "Some WASM extensions are NOT available, falling back to the vanilla WASM module"
+        );
+    }
+
     __webpack_public_path__ = publicPath(config);
+
+    // Note: The argument passed to import() has to be a simple string literal,
+    // otherwise some bundler will get confused and won't include the module?
+    const {
+        default: init,
+        Ruffle,
+        isWasmSimdUsed,
+    } = await (extensionsSupported
+        ? import("../pkg/ruffle_web_fancy")
+        : import("../pkg/ruffle_web"));
+
     await init();
 
+    if (extensionsSupported && !isWasmSimdUsed()) {
+        console.log("The loaded fancy WASM module is a fake one!");
```

Comment:
Maybe we could do that in the same spot we do
```
console.log("New Ruffle instance created.");
```
?
This way we logspam a bit less.
For example:
```
New Ruffle instance created (wasm features: on).
New Ruffle instance created (wasm features: off).
```

Reply:
Hm, I like the idea, but then I'd either have to return `isWasmSimdUsed` from `fetchRuffle` and then from `loadRuffle`, or make it a member of `Ruffle`. Is that right?

Reply:
> or make it a member of Ruffle

Yeah, I don't see anything wrong with that :)